worlds/gl/__init__.py

```diff
@@ -169,60 +155,49 @@ def create_items(self) -> None:
         # First add in all progression and useful items
         required_items = []
         precollected = [item for item in item_list if item in self.multiworld.precollected_items[self.player]]
-        for item in item_list:
-            if item.progression != ItemClassification.filler and item.progression != ItemClassification.trap and item not in precollected:
-                if "Obelisk" in item.item_name and self.options.obelisks == 0:
-                    continue
-                if "Mirror" in item.item_name and self.options.mirror_shards == 0:
-                    continue
+        skipped_items = [item for item in item_list if ("Key" in item.item_name and self.options.infinite_keys)
+                                                        or ("Boots" in item.item_name and self.options.permanent_speed)
+                                                        or ("Poison" in item.item_name and (self.options.traps_choice == "only_death" or self.options.traps_choice == "none_active"))
+                                                        or ("Death" in item.item_name and (self.options.traps_choice == "only_fruit" or self.options.traps_choice == "none_active"))
+                                                        or ("Obelisk" in item.item_name and self.options.obelisks == 0)
+                                                        or ("Mirror" in item.item_name and self.options.mirror_shards == 0)]
+        for item in [item_ for item_ in item_list if ItemClassification.progression in item_.progression or ItemClassification.useful in item_.progression and item_ not in precollected and item_ not in skipped_items]:
```

Comment:
FYI in this general section you go over the style guide's 120 character limit a few times, and while it is a guide and not a requirement breaking comprehensions up like this is pretty simple and can make them easier to read too
```suggestion
        for item in [item_ for item_ in item_list
                     if ItemClassification.progression in item_.progression
                     or ItemClassification.useful in item_.progression
                     and item_ not in precollected
                     and item_ not in skipped_items
                     ]:
```